apis/scheduling/v1alpha1/types.go

```diff
@@ -47,13 +47,15 @@ type ElasticQuota struct {
 // ElasticQuotaSpec defines the Min and Max for Quota.
 type ElasticQuotaSpec struct {
 	// Min is the set of desired guaranteed limits for each named resource.
+	// +kubebuilder:default={cpu:0,memory:0}
 	// +optional
-	Min v1.ResourceList `json:"min,omitempty" protobuf:"bytes,1,rep,name=min, casttype=ResourceList,castkey=ResourceName"`
+	Min v1.ResourceList `json:"min" protobuf:"bytes,1,rep,name=min, casttype=ResourceList,castkey=ResourceName"`
 
 	// Max is the set of desired max limits for each named resource. The usage of max is based on the resource configurations of
 	// successfully scheduled pods.
+	// +kubebuilder:default={cpu:9223372036854775807,memory:9223372036854775807}
```

Comment:
what about other resources? Like `gpu`, hugepages, ephemeral-storage or devices advertised by device plugins?

Reply:
Thank you for your comment.

In my opinion, I don't think it's necessary to cover all resources. One reason is that since new resources may be added in the future, it would be difficult to accommodate them all. For example, for the GPU plugin, there are nvidia.com/gpu and amd.com/gpu, and more may be added in the future(https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/). It's not necessary to research and cover all of them.

Another reason is that there are limits to the fields that the Resource type can hold. Max/min is converted to ElasticQuotaInfo as a type Resource(https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/types.go#L432-L441). The fields that the Resource type explicitly has are only MilliCPU, Memory, EphemeralStorage, and AllowedPodNumber (AllowedPodNumber is not related to ElasticQuota), so I don't think it's necessary to declare fields other than cpu, memory, and ephemeral-storage. And other fields are handled as ScalarResources(https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/pkg/capacityscheduling/elasticquota.go#L186), so I don't use specific name other than CPU, Memory and EphemeralStorage.

However, while writing this reply, I thought that it may be okay to add ephemeral-storage, so I will include it.

Please let me know if there is anything wrong with my response.